core/src/main/java/hudson/PluginManager.java

```diff
@@ -1742,11 +1788,21 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
         try {
             Jenkins.get().checkPermission(Jenkins.ADMINISTER);
 
+            String fileName = "";
+            PluginCopier copier;
             ServletFileUpload upload = new ServletFileUpload(new DiskFileItemFactory());
+            List<FileItem> items = upload.parseRequest(req);
+            if(StringUtils.isNotBlank(items.get(1).getString())) {
```

Comment:
is it safe to do a `items.get(1)` without any length checks?

Reply:
Previously the code was doing .get(0) with no length check. Each form field creates and item and then there are any crumbs, etc. I was thinking about reworking to use stream and look for the field names, but followed the existing method.